hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/storefiletracker/StoreFileTrackerFactory.java

```diff
@@ -311,4 +312,40 @@ public static void checkForModifyTable(Configuration conf, TableDescriptor oldTa
       }
     }
   }
+
+  /**
+   * Makes sure restoring a snapshot does not break the current SFT setup
+   * follows StoreUtils.createStoreConfiguration
+   * @param currentTableDesc Existing Table's TableDescriptor
+   * @param snapshotTableDesc Snapshot's TableDescriptor
+   * @param baseConf Current global configuration
+   * @throws RestoreSnapshotException if restore would break the current SFT setup
+   */
+  public static void checkForRestoreSnapshot(TableDescriptor currentTableDesc,
```

Comment:
nit: maybe `validatePreRestoreSnapshot(..)` to better indicate what we're doing (validating the state) and when we are doing it (before a restore_snapshot).

"Validate" is a bit more specific than "check" to me, but I recognize this is also subjective.